src/nvim/regexp_nfa.c

```diff
@@ -4853,6 +4854,9 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
   fast_breakcheck();
   if (got_int)
     return FALSE;
+  if (nfa_time_limit != NULL && profile_passed_limit(*nfa_time_limit)) {
+    return FALSE;
```

Comment:
Understood. My last comment was a general comment regarding the alternative (i.e. "leave the old code alone"). 

It doesn't seem to me any worse to assign `false` to `int` than `FALSE`. Because the implied semantics of `FALSE` and `TRUE` are just as wrong as `false` and `true`. A tri-state should be an enum, ideally.